### PR TITLE
update search script for newer Watson service

### PR DESCRIPTION
**Pull request by dhmlau**
- Update the `search.js` to use the latest Watson Discovery service. 
- Update deployment section of README to include instructions on how to set things up.

### CODE REVIEW

File: search.js [hunk 1]
```diff
@@ -98,12 +100,12 @@ function main(params) {
             x = `metadata.sidebar:"${x}"`;
           });
           filter = `(${sidebars.join('|')})`;
-          data.filter = filter;
+        // data.filter = filter;
```
Comment (jannyHou) [inline search.js]: is it intended to ignore the filter?

Comment (dhmlau) [reply search.js]: I wasn't able to get the search to return anything if this line is running. But currently we have the problem that it always return you the LB4 results first even you're searching from a LB3 page.

Comment (dhmlau) [reply search.js]: Looks like the previous way to generate a `posts.json` file for the data content has deprecated:
https://cloud.ibm.com/docs/services/discovery?topic=discovery-getting-started-with-the-data-crawler. I think the filtering by sidebar value no longer works for us.  I'll update the PR. Thanks.

**Review by raymondfeng — APPROVED** [review]

**Review by jannyHou — APPROVED** [review]
:shipit: Cool LGTM.

**Pull request merged**